aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -142,27 +142,36 @@ impl IndyCredxAnonCreds {
     async fn _get_credentials_for_proof_req_for_attr_name(
         &self,
         restrictions: Option<&Value>,
-        attr_name: &str,
+        attr_names: Vec<String>,
     ) -> VcxCoreResult<Vec<(String, CredxCredential)>> {
-        let attr_marker_tag_name = _format_attribute_as_marker_tag_name(attr_name);
+        let mut attrs = Vec::new();
 
-        let wql_attr_query = json!({
-            attr_marker_tag_name: "1"
-        });
+        for name in attr_names {
+            let attr_marker_tag_name = _format_attribute_as_marker_tag_name(&name);
+
+            let wql_attr_query = json!({
+                attr_marker_tag_name: "1"
+            });
+
+            attrs.push(wql_attr_query);
+        }
 
         let restrictions = restrictions.map(|x| x.to_owned());
 
         let wql_query = if let Some(restrictions) = restrictions {
             match restrictions {
                 Value::Array(mut arr) => {
-                    arr.push(wql_attr_query);
+                    arr.extend(attrs);
                     json!({ "$and": arr })
                 }
-                Value::Object(obj) => json!({ "$and": vec![wql_attr_query, Value::Object(obj)] }),
-                _ => wql_attr_query,
+                Value::Object(obj) => {
+                    attrs.push(Value::Object(obj));
+                    json!({ "$and": attrs })
+                }
+                _ => json!(attrs),
             }
         } else {
-            wql_attr_query
+            json!(attrs)
```

Comment:
I think in these cases, we'd still want things to be ANDed together;

I.e. the query is for where 'attr1=="1" && attr2=="2" && attr3=="3" etc'. So I think it's still json({"$and": attrs}).

Either way, these branches are a bit of an edge case. I might make ticket about the wallet queries in this class, we really should be using the Indy-wql crate, or create our own utility for query construction